app/code/Magento/QuoteGraphQl/Model/Resolver/CartItem/CustomizableOptions.php

```diff
@@ -0,0 +1,187 @@
+            return $this->valueFactory->create(function () {
+                return [];
+            });
+        }
+
+        /** @var QuoteItem $cartItem */
+        $cartItem = $value['model'];
+        $optionIds = $cartItem->getOptionByCode('option_ids');
+
+        if (!$optionIds) {
+            return $this->valueFactory->create(function () {
+                return [];
+            });
+        }
+
+        $customOptions = [];
+        $customOptionIds = explode(',', $optionIds->getValue());
+
+        foreach ($customOptionIds as $optionId) {
+            $customOptionData = $this->getOptionData($cartItem, (int) $optionId);
+
+            if (0 === count($customOptionData)) {
+                continue;
+            }
+
+            $customOptions[] = $customOptionData;
+        }
+
+        $result = function () use ($customOptions) {
+            return $customOptions;
+        };
+
+        return $this->valueFactory->create($result);
+    }
+
+    /**
+     * @param QuoteItem $cartItem
+     * @param int $optionId
+     * @return array
+     */
+    private function getOptionData($cartItem, int $optionId): array
+    {
+        $product = $cartItem->getProduct();
+        $option = $product->getOptionById($optionId);
+
+        if (!$option) {
+            return [];
+        }
+
+        $itemOption = $cartItem->getOptionByCode('option_' . $option->getId());
+
+        /** @var SelectOptionType|TextOptionType|DefaultOptionType $optionTypeGroup */
+        $optionTypeGroup = $option->groupFactory($option->getType())
+            ->setOption($option)
+            ->setConfigurationItem($cartItem)
+            ->setConfigurationItemOption($itemOption);
+
+        if (ProductCustomOptionInterface::OPTION_GROUP_FILE == $option->getType()) {
+            $downloadParams = $cartItem->getFileDownloadParams();
+
+            if ($downloadParams) {
+                $url = $downloadParams->getUrl();
+                if ($url) {
+                    $optionTypeGroup->setCustomOptionDownloadUrl($url);
+                }
+                $urlParams = $downloadParams->getUrlParams();
+                if ($urlParams) {
+                    $optionTypeGroup->setCustomOptionUrlParams($urlParams);
+                }
+            }
+        }
+
+        $selectedOptionValueData = [
+            'id' => $itemOption->getId(),
+            'label' => $optionTypeGroup->getFormattedOptionValue($itemOption->getValue()),
+        ];
+
+        if (ProductCustomOptionInterface::OPTION_TYPE_DROP_DOWN == $option->getType()
+            || ProductCustomOptionInterface::OPTION_TYPE_RADIO == $option->getType()
+            || ProductCustomOptionInterface::OPTION_TYPE_CHECKBOX == $option->getType()
+        ) {
+            $optionValue = $option->getValueById($itemOption->getValue());
+            $selectedOptionValueData['price'] = [
+                'type' => strtoupper($optionValue->getPriceType()),
+                'units' => '$',
```

Comment:
I believe this part, including files processing will be completed before we merge.

Reply:
Yes, I have been implemented this mock. Thank you for noticing!